src/controller/stream-controller.ts

```diff
@@ -769,7 +769,37 @@ export default class StreamController
               3,
             )}`,
           );
-          media.currentTime = liveSyncPosition;
+
+          if (this.config.liveSyncMode === 'buffered') {
+            const bufferInfo = BufferHelper.bufferInfo(
+              this.media,
+              liveSyncPosition,
+              0,
+            );
+
+            if (!bufferInfo || bufferInfo.buffered?.length === 0) {
```

Comment:
Don't strictly compare to `0` when `buffered?` could be undefined:

```suggestion
            if (!bufferInfo || !bufferInfo.buffered?.length) {
```